pkgs/os-specific/bsd/netbsd/pkgs/libc.nix

```diff
@@ -1,102 +1,52 @@
 {
   lib,
-  mkDerivation,
-  defaultMakeFlags,
-  _mainLibcExtraPaths,
-  bsdSetupHook,
-  netbsdSetupHook,
-  makeMinimal,
-  install,
-  mandoc,
-  groff,
-  flex,
-  byacc,
-  genassym,
-  gencat,
-  lorder,
-  tsort,
-  statHook,
-  rsync,
-  rpcgen,
-  csu,
-  headers,
+  symlinkJoin,
+  libc-minimal,
+  libpthread,
+  libm,
+  libresolv,
+  librpcsvc,
+  i18n_module,
+  libutil,
   librt,
+  libcrypt,
+  version,
 }:
 
-mkDerivation {
-  noLibc = true;
-  path = "lib/libc";
-  USE_FORT = "yes";
-  MKPROFILE = "no";
-  extraPaths = _mainLibcExtraPaths ++ [ "external/bsd/jemalloc" ];
-  nativeBuildInputs = [
-    bsdSetupHook
-    netbsdSetupHook
-    makeMinimal
-    install
-    mandoc
-    groff
-    flex
-    byacc
-    genassym
-    gencat
-    lorder
-    tsort
-    statHook
-    rsync
-    rpcgen
-  ];
-  buildInputs = [
-    headers
-    csu
-  ];
-  env.NIX_CFLAGS_COMPILE = "-B${csu}/lib -fcommon";
-  meta.platforms = lib.platforms.netbsd;
-  SHLIBINSTALLDIR = "$(out)/lib";
-  MKPICINSTALL = "yes";
-  NLSDIR = "$(out)/share/nls";
-  makeFlags = defaultMakeFlags ++ [ "FILESDIR=$(out)/var/db" ];
-  postInstall = ''
-    pushd ${headers}
-    find . -type d -exec mkdir -p $out/\{} \;
-    find . \( -type f -o -type l \) -exec cp -pr \{} $out/\{} \;
-    popd
-
-    pushd ${csu}
-    find . -type d -exec mkdir -p $out/\{} \;
-    find . \( -type f -o -type l \) -exec cp -pr \{} $out/\{} \;
-    popd
-
-    NIX_CFLAGS_COMPILE+=" -B$out/lib"
-    NIX_CFLAGS_COMPILE+=" -I$out/include"
-    NIX_LDFLAGS+=" -L$out/lib"
-
-    make -C $BSDSRCDIR/lib/libpthread $makeFlags
-    make -C $BSDSRCDIR/lib/libpthread $makeFlags install
-
-    make -C $BSDSRCDIR/lib/libm $makeFlags
-    make -C $BSDSRCDIR/lib/libm $makeFlags install
-
-    make -C $BSDSRCDIR/lib/libresolv $makeFlags
-    make -C $BSDSRCDIR/lib/libresolv $makeFlags install
+symlinkJoin rec {
+  name = "${pname}-${version}";
+  pname = "libc-netbsd";
+  inherit version;
 
-    make -C $BSDSRCDIR/lib/librpcsvc $makeFlags
-    make -C $BSDSRCDIR/lib/librpcsvc $makeFlags install
-
-    make -C $BSDSRCDIR/lib/i18n_module $makeFlags
-    make -C $BSDSRCDIR/lib/i18n_module $makeFlags install
-
-    make -C $BSDSRCDIR/lib/libutil $makeFlags
-    make -C $BSDSRCDIR/lib/libutil $makeFlags install
-
-    make -C $BSDSRCDIR/lib/librt $makeFlags
-    make -C $BSDSRCDIR/lib/librt $makeFlags install
+  outputs = [
+    "out"
+    "dev"
+    "man"
```

Comment:
Is symlinkJoin smart enough to actually use all three of these outputs?

Reply:
Yes!